crates/accelerate/src/commutation_cancellation.rs

```diff
@@ -0,0 +1,285 @@
+                                }
+                                if num_qargs == 1usize && _x_rotations.contains(op_name.as_str()) {
+                                    single_q_cancellation_sets
+                                        .entry((X_ROTATION.to_string(), wire, com_set_idx))
+                                        .or_insert_with(Vec::new)
+                                        .push(*node);
+                                }
+                                // Don't deal with Y rotation, because Y rotation doesn't commute with
+                                // CNOT, so it should be dealt with by optimized1qgate pass
+                                if num_qargs == 2usize
+                                    && dag.get_qargs(op.qubits).first().unwrap() == &wire
+                                {
+                                    let second_qarg = dag.get_qargs(op.qubits)[1];
+                                    let q2_key = (
+                                        op_name,
+                                        wire,
+                                        second_qarg,
+                                        com_set_idx,
+                                        *node_indices
+                                            .get(&(*node, Wire::Qubit(second_qarg)))
+                                            .unwrap(),
+                                    );
+                                    two_q_cancellation_sets
+                                        .entry(q2_key)
+                                        .or_insert_with(Vec::new)
+                                        .push(*node);
+                                }
+                            }
+                        })
+                    }
+                })
+        }
+    });
+
+    for (cancel_key, cancel_set) in &two_q_cancellation_sets {
+        if cancel_set.len() > 1 && _gates.contains(cancel_key.0.as_str()) {
+            for &c_node in &cancel_set[0..(cancel_set.len() / 2) * 2] {
+                dag.remove_op_node(c_node);
+            }
+        }
+    }
+
+    for (cancel_key, cancel_set) in &single_q_cancellation_sets {
+        if cancel_key.0 == Z_ROTATION && z_var_gate.is_none() {
+            continue;
+        }
+        if cancel_set.len() > 1 && _gates.contains(cancel_key.0.as_str()) {
+            for &c_node in &cancel_set[0..(cancel_set.len() / 2) * 2] {
+                dag.remove_op_node(c_node);
+            }
+        } else if cancel_set.len() > 1 && (cancel_key.0 == Z_ROTATION || cancel_key.0 == X_ROTATION)
+        {
+            let run_op = match &dag.dag[*cancel_set.first().unwrap()] {
+                NodeType::Operation(instr) => instr,
+                _ => panic!("Unexpected type in commutation set run."),
+            };
+
+            let run_qarg = dag.get_qargs(run_op.qubits).first().unwrap();
+            let mut total_angle: f64 = 0.0f64;
+            let mut total_phase: f64 = 0.0f64;
```

Comment:
You don't need the suffix because you've set a type annotation here.
```suggestion
            let mut total_angle: f64 = 0.0;
            let mut total_phase: f64 = 0.0;
```